File: javascript/aws-lambda/security/pg-sqli.js
let response;

const prettyPrint = (ob) => JSON.stringify(ob, null, 2).replace('\'','');
const timestamp = () => new Date();

const toBase64 = (msg) => Buffer.from(msg).toString('base64');

const { Client } = require('pg');
exports.handler = async function (event, context) {
  console.log("the event");
  console.log(event);

  const client = new Client({
    user: "test",
    host: "db",
    database: "nockslots",
    password: "test",
    port: 5432,
  });

  console.log("connecting to db...");

  await client.connect();

  records = [];
  event.Records.forEach((record) => {
    const { body } = record;
    console.log(body);
    records.push(toBase64(body));
  });

  const query = `INSERT INTO public.messages (body, encoded_message) VALUES ('${prettyPrint(event)}', '${records[0]}');`;

  console.log('the query:');
  console.log(query);

  try {
    console.log("Trying the query...");
    // ruleid: pg-sqli
    await client.query(query)
    // ok: pg-sqli
    await client.query('INSERT INTO messages (body, message) VALUES ($1, $2);', [prettyPrint(event), records[0]])
    await client.end();
  } catch (error) {
    console.log('Could not add row to postgres, soz');
    console.log(error);
  }

  return { key: JSON.stringify(records) };
};

File: ruby/aws-lambda/security/pg-sqli.rb
require 'pg'

def show(event:, context:)
  conn = PG::Connection.open(:dbname => 'test')

  # ok: pg-sqli
  res = conn.exec_params('SELECT $1 AS a, $2 AS b, $3 AS c', [event['id'], 2, nil])

  # ruleid: pg-sqli
  res2 = conn.exec_params('SELECT * FROM foobar WHERE id = %{id}' % {id: event['id']})

  {
    body: res2
  }
end